### PR TITLE
first post about hypervisor development

### PR DESCRIPTION
**Pull request by krystian-hebel**

### CODE REVIEW

File: _posts/2019-03-XX-5-terms-every-hypervisor-developer-should-know.md [hunk 1]
```diff
@@ -0,0 +1,292 @@
+---
+post_title: 5 terms every hypervisor developer should know
```
Comment (pietrushnic) [inline _posts/2019-03-XX-5-terms-every-hypervisor-developer-should-know.md]: @krystian-hebel I like this title very much :)

File: _posts/2019-03-XX-5-terms-every-hypervisor-developer-should-know.md [hunk 2]
```diff
@@ -0,0 +1,292 @@
...
+Virtual machine, also called guest software. It consists of OS and application
+software. Software running on VM usually is not aware of the fact that VMM
+controls hardware and that there may be other VMs running on the same platform,
+but it can be. Most of the times any application that can run on real hardware
+can also run inside a VM, depending on hypervisor implementation.
+
+As with VMM, VMs can also be symmetrical or asymmetrical when it comes to
+multi-processor execution. Again, I will restrict VM to one core. Here it makes
+even more sense because VMM can implement a scheduler and run multiple VMs on
+one physical core, in turn. For most people that were using only type-2
```
Comment (pietrushnic) [inline _posts/2019-03-XX-5-terms-every-hypervisor-developer-should-know.md]: @krystian-hebel maybe some basic explanation about types and examples of given types?

File: _posts/2019-03-XX-5-terms-every-hypervisor-developer-should-know.md [hunk 3]
```diff
@@ -0,0 +1,292 @@
...
+   are proper for supporting VMX non-root operation and that the VMCS is
+   correctly configured to support the next VM exit.
+3. The guest-state area of the VMCS is checked to ensure that, after the VM
+   entry completes, the state of the logical processor is legal.
+4. MSRs are loaded from the VM-entry MSR-load area.
+5. If VMLAUNCH is being executed, the launch state of the VMCS is set to
+   "launched".
+6. An event may be injected in the guest context.
+
+#### VM exits
```
Comment (pietrushnic) [inline _posts/2019-03-XX-5-terms-every-hypervisor-developer-should-know.md]: @krystian-hebel some diagram in draw.io would be great - nothing fancy, but something that present idea

File: _posts/2019-03-XX-5-terms-every-hypervisor-developer-should-know.md [hunk 4]
```diff
@@ -48,6 +51,25 @@ to get some statistics from hypervisor - useful for debugging. Without these
...
+  the hypervisor itself, are just other processes running on top of the
+  operating system. Examples are [VMware Workstation](https://www.vmware.com/products/workstation-pro.html),
+  [VMware Player](https://www.vmware.com/products/workstation-player/workstation-player-evaluation.html),
+  [VirtualBox](https://www.virtualbox.org/) and [QEMU](https://www.qemu.org/).
+
+Note that there is no clear distinction between those two, there are some
+hypervisors that are somewhere in between. [KVM](https://www.linux-kvm.org/page/Main_Page)
+for example is a kernel module that converts a whole OS to a type-1 hypervisor,
+but because Linux still is an operating system it has to control access to
+resources between processes, as type-2 hypervisor does.
```
Comment (pietrushnic) [inline _posts/2019-03-XX-5-terms-every-hypervisor-developer-should-know.md]: @krystian-hebel and Bareflank claims it is type-0 :)

**Comment by miczyg1** [comment]
@krystian-hebel could You please address all remaining discussions?

File: _posts/2019-03-XX-5-terms-every-hypervisor-developer-should-know.md (hunk 2 above)
Comment (krystian-hebel) [reply _posts/2019-03-XX-5-terms-every-hypervisor-developer-should-know.md]: done

File: _posts/2019-03-XX-5-terms-every-hypervisor-developer-should-know.md (hunk 3 above)
Comment (krystian-hebel) [reply _posts/2019-03-XX-5-terms-every-hypervisor-developer-should-know.md]: @pietrushnic I'm not sure if this is good idea, there is too much text to make it readable

Comment (pietrushnic) [reply _posts/2019-03-XX-5-terms-every-hypervisor-developer-should-know.md]: @krystian-hebel as you wish, that was just an idea.

File: _posts/2019-03-XX-5-terms-every-hypervisor-developer-should-know.md [hunk 5]
```diff
@@ -0,0 +1,268 @@
...
+post_date: 2019-03-DD HH:MM:SS
+
+tags:
+    - hypervisor
+    - virtualization
+    - bare-metal
+    - VMX
+categories:
+    - Firmware
+
```
Comment (artur-rs) [inline _posts/2019-03-XX-5-terms-every-hypervisor-developer-should-know.md]: Outdated header

File: _posts/2019-03-XX-5-terms-every-hypervisor-developer-should-know.md [hunk 6]
```diff
@@ -0,0 +1,268 @@
+---
+post_title: 5 terms every hypervisor developer should know
+author: Krystian Hebel
+layout: post
+published: true
+post_date: 2019-03-DD HH:MM:SS
```
Comment (artur-rs) [inline _posts/2019-03-XX-5-terms-every-hypervisor-developer-should-know.md]: post is scheduled on 2019-04-30

File: _posts/2019-03-XX-5-terms-every-hypervisor-developer-should-know.md [hunk 7]
```diff
@@ -0,0 +1,268 @@
...
+control fields are sometimes referred to collectively as VMX controls.
+
+An overview of fields available in VMCS is available [here](https://github.com/LordNoteworthy/cpu-internals/raw/master/VMCS.pdf).
+Note that not all of those fields are available on all processors, also new ones
+can be added in the future.
+
+VMCS can be in one of six launch states, depending mostly on the last
+VMCS-related instruction called:
+
+![VMCS states](https://3mdeb.com/wp-content/uploads/2019/04/vmcs_states.png)
```
Comment (artur-rs) [inline _posts/2019-03-XX-5-terms-every-hypervisor-developer-should-know.md]: link is dead

**Comment by artur-rs** [comment]
@krystian-hebel parent branch should be set to `develop`

File: _posts/2019-03-XX-5-terms-every-hypervisor-developer-should-know.md (hunk 5 above)
Comment (krystian-hebel) [reply _posts/2019-03-XX-5-terms-every-hypervisor-developer-should-know.md]: fixed

File: _posts/2019-03-XX-5-terms-every-hypervisor-developer-should-know.md (hunk 6 above)
Comment (krystian-hebel) [reply _posts/2019-03-XX-5-terms-every-hypervisor-developer-should-know.md]: done

File: _posts/2019-03-XX-5-terms-every-hypervisor-developer-should-know.md (hunk 7 above)
Comment (krystian-hebel) [reply _posts/2019-03-XX-5-terms-every-hypervisor-developer-should-know.md]: is it proper path now?

Comment (artur-rs) [reply _posts/2019-03-XX-5-terms-every-hypervisor-developer-should-know.md]: @krystian-hebel Looks good. You can always check if the post is rendering correctly by setting local server - step by step instruction is located in `README.md` file

**Pull request merged**